cmd/demo/main.go

```diff
@@ -11,132 +12,156 @@ import (
 	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-12-01/compute"
 	"github.com/Azure/go-autorest/autorest/adal"
 	"github.com/Azure/go-autorest/autorest/azure/auth"
-	"github.com/spf13/pflag"
 	"k8s.io/klog/v2"
 )
 
+const (
+	contextTimeout = 80 * time.Second
+)
+
 var (
-	retryWaitTime  = pflag.Int("retry-wait-time", 20, "retry wait time in seconds")
-	resource       = pflag.String("aad-resourcename", "https://management.azure.com/", "name of resource to grant token")
-	subscriptionID = pflag.String("subscriptionid", "", "subscription id for test")
-	clientID       = pflag.String("clientid", "", "client id for the msi id")
-	resourceGroup  = pflag.String("resourcegroup", "", "any resource group name with reader permission to the aad object")
+	period           time.Duration
+	resourceName     string
+	subscriptionID   string
+	resourceGroup    string
+	identityClientID string
 )
 
 func main() {
-	pflag.Parse()
+	flag.DurationVar(&period, "period", 100*time.Second, "The period that the demo is being executed")
```

Comment:
This is a breaking change for the demo image. I think AKS documents the usage and logs in the docs as well: https://docs.microsoft.com/en-us/azure/aks/use-azure-ad-pod-identity. 

cc @miwithro @bcho to update docs with demo image after we cut next release of pod identity.